lib/more_core_extensions/core_ext/enumerable/sorting.rb

```diff
@@ -0,0 +1,39 @@
+module MoreCoreExtensions
+  module StableSorting
+    def self.included(klass)
+      klass.class_eval do
```

Comment:
This is a weird notation, and ends up modifying the original class instead of adding a module into the ancestors chain (the latter is always how we do it in this gem).  I think it's preferable to do:

```ruby
module MoreCoreExtensions
  module StableSorting
    def stable_sort_by...
    end
  end
end

Enumerable.extend(MoreCoreExtensions::StableSorting)
```

Reply:
yes I know @Fryguy but  because `Enumerable` is as module included to `Array` class when I will use our typical way (`Enumerable.extend(MoreCoreExtensions::StableSorting)...`) the newly added method `stable_sort_by` is not applicable on `Array` instances, it didn't reflect this extension:

```
irb(main):007:0> [].stable_sort_by
NoMethodError: undefined method `stable_sort_by' for []:Array
	from (irb):7
```

1) so because of that I have to change original `Enumerable` or is here any other solution ? (and in order to don't not use this _weird notation_)

2) or I can include this extension to Array separately...

Reply:
Ah ok...I've been trying to experiment locally and I can't get it to work either, so for now this is fine.

Reply:
@lpichler Is there any particular reason this is on Enumerable and not just Array directly?  I can't see use cases for patching anything other than Array, particularly because this method requires an Array of Hashes specifically to even work.  It might make it easier to patch if it's just Array.

Reply:
@Fryguy you are right, [Rupport::Data::Table](https://github.com/ManageIQ/manageiq/blob/e1b321251d05bd6d94ffd2f3fba2101a33a72c93/lib/patches/ruport_patch.rb#L12) has `Enumerable` but `to_a` is used in this method. (so it doesn't matter if I will use `to_a` on  `Rupport::Data::Table` before calling [stable_sort_by](https://github.com/ManageIQ/manageiq/pull/18389/commits/a932307515bf89b4efe8040170591ef0afbac97f#diff-acaa15c6f37e2af4c88991523fc037c0R9)). In other PR I am working with `Array`.

So I will create PR where:
- this method is moved to `Array`
- this method is renamed to `tabular_sort`